### ISSUE TITLE
Data are downloaded on every get request

### ISSUE THREAD

**Issue by blaumeik51** [FL550/simple_dwd_weatherforecast#1]
I get the Impression, that the kml file ist fatched on every get query (I am fetching one hour after another in a outer loop and it takes about 2s per hour). From reading the code (I am Python beginner) I understood, that new data shall be fatched only if the kml issued time is older than 6 hours. Sometimes running the same code goes very fast. So I assume that there ist an issue with the issue time supplied by the dwd. Is there any solution to it?

**Comment by FL550** [comment]
Thanks for the report! You are right, I read the issue time from the kml file and try to limit the requests to the server.

I will look into this issue at friday. Can you provide me the station id you are using?

**Comment by blaumeik51** [comment]
Hey, yes sure. It is 10147, but I had the same issue with other stations as well. At 6:15 in the morning it goes fast as expected for some reason. 
Do you know the issue times? Is there a way to get back the entire result at one shot? Like giving a list of dates?
Thanks for your help.
Meik

**Comment by FL550** [comment]
I hadn't much testing time after releasing the package, so I am not sure about the reliability of the issue time. If the issue time in the KML file is not correct, this might causing the constant reload.

They issue new forecasts every 6 hours, thus I added the corresponding delay. 
I can implement a new way of checking for new forecast data by looking at the available files at the server. This solution avoids actually downloading the kml file if no new data is available. I can't say if this is more reliable, but it's worth trying.

**Comment by blaumeik51** [comment]
It depents... In order to get the full data set one needs to call the get functions 240 times. Checking each time for new data on the server seems not ideal either. Maybe you can remember the issue time or check for the current hour as the data seem to be updated at 00, 06, 12, 18 UTC and call for new data only if the fetched data are older.

**Comment by FL550** [comment]
So by getting the full dataset you mean getting data for the next 10 days for **one** station, right? 

If you are getting data for 240 **stations** this is normal behavior as the forecast is provided in separate files per station.

**Comment by FL550** [comment]
Reviewing the code I saw that I am currently not respecting the timezone. So this might cause some unintended get request to the server. But in your case this is only 8 times a day.

**Comment by blaumeik51** [comment]
Yes I mean getting the full dataset for one station, i.e. the forecast for ten days in one shot would be great. I guess the kml file contains already the full data set.

I tried pulling data during the day at various times. In many cases it works well. So it might be that the time zone issue solves it. Are you pausing for six hours after the request or the issue time? 

Is the input time to the get functions supposed to be utc or german local time?

**Comment by FL550** [comment]
The Class variable `forecast_data` already contains the complete forecast as a dictionary. You can access it like this:

```
dwd_weather​ ​=​ ​dwdforecast​.​Weather​(​"10385"​) # Station-ID For BERLIN-SCHOENEFELD​
dwd_weather.update()
access_forecast_dict = dwd_weather.forecast_data
```

I am pausing for 6 hours after the issue time and the input time is supposed to be in UTC.

**Comment by blaumeik51** [comment]
Yeah that helps. However, the condition is not given correctly. It is either empty or a single digit number string. I was expecting here the same string as with the get_forecast_condition method. Maybe there is a bug as well. 
Anyway, the package helps me a lot, so thank you very much for your efford.

**Comment by FL550** [comment]
The digit is the original weather code provided by DWD, so it's not a bug. I only translate and summarize them to a human readable format. But as this is done in a very subjective way, I didn't want to replace the original info. 

Feel free to use the translation I use by copying that part of my source code. If you want to translate it yourself, the detailed list of DWD can be found here: https://www.dwd.de/DE/leistungen/opendata/help/schluessel_datenformate/kml/mosmix_element_weather_xls.xlsx?__blob=publicationFile&v=4

I'm happy you find my library useful and thanks for guiding me to the timezone-bug! And maybe I should also update the readme 🤔😊

**Comment by FL550** [comment]
The UTC bug is fixed in 4c625df and the README is updated.
